Signal/src/ViewControllers/ConversationView/Cells/MediaGalleryCellView.swift

```diff
@@ -0,0 +1,219 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+@objc(OWSMediaGalleryCellView)
+public class MediaGalleryCellView: UIStackView {
+    private let items: [ConversationMediaGalleryItem]
+    private let itemViews: [ConversationMediaView]
+
+    private static let kSpacingPts: CGFloat = 2
+    private static let kMaxItems = 5
+
+    @objc
+    public required init(mediaCache: NSCache<NSString, AnyObject>,
+                         items: [ConversationMediaGalleryItem],
+                         maxMessageWidth: CGFloat) {
+        self.items = items
+        self.itemViews = MediaGalleryCellView.itemsToDisplay(forItems: items).map {
+            ConversationMediaView(mediaCache: mediaCache,
+                          attachment: $0.attachment)
+        }
+
+        super.init(frame: .zero)
+
+        backgroundColor = Theme.backgroundColor
+
+        createContents(maxMessageWidth: maxMessageWidth)
+    }
+
+    private func createContents(maxMessageWidth: CGFloat) {
+        switch itemViews.count {
+        case 0:
+            return
+        case 1:
+            guard let itemView = itemViews.first else {
+                owsFailDebug("Missing item view.")
+                return
+            }
+            addSubview(itemView)
+            itemView.autoPinEdgesToSuperviewEdges()
+        case 4:
```

Comment:
no biggie, but is there a reason our case statements aren't increasing monotonically?

Reply:
Done.  I was trying to align with how the (very similar) measurement logic is laid out, and the 1 and 4 cases are treated identically there.